### PR TITLE
Fix API /content/history is very slow with big sites

### PR DESCRIPTION
**Pull request by phuongnq**
https://github.com/craftersoftware/craftercms/issues/1052

### CODE REVIEW

File: src/main/resources/crafter/studio/database/upgrade/4.2.x/4.2.0.15-to-4.2.0.16.sql [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+ALTER TABLE `audit`
+    ADD INDEX `audit_primary_target_value_idx` (`primary_target_value`) ;
```
Comment (jmendeza) [inline src/main/resources/crafter/studio/database/upgrade/4.2.x/4.2.0.15-to-4.2.0.16.sql:18]: @phuongnq have we tested the impact of this index in the insertion performance?

File: src/main/resources/org/craftercms/studio/api/v2/dal/AuditDAO.xml [hunk 2]
```diff
@@ -280,6 +285,18 @@
         AND origin = 'API'
     </select>
 
+    <select id="getCommitAuthors" parameterType="java.util.Map" resultMap="CommitAuthorMap">
+        SELECT a.commit_id, u.*
+        FROM audit a
+        INNER JOIN user u on u.username = a.actor_id
+        WHERE a.commit_id IN
```
Comment (jmendeza) [inline src/main/resources/org/craftercms/studio/api/v2/dal/AuditDAO.xml]: Should we filter by site id? Can we have the same commit id in different sites if cloning from the same remote?

**Review by jmendeza — CHANGES_REQUESTED** [review]

File: src/main/resources/org/craftercms/studio/api/v2/dal/AuditDAO.xml (hunk 2 above)
Comment (phuongnq) [reply src/main/resources/org/craftercms/studio/api/v2/dal/AuditDAO.xml]: @jmendeza It's best to filter with the site id so I updated to include this

File: src/main/resources/crafter/studio/database/upgrade/4.2.x/4.2.0.15-to-4.2.0.16.sql (hunk 1 above)
Comment (phuongnq) [reply src/main/resources/crafter/studio/database/upgrade/4.2.x/4.2.0.15-to-4.2.0.16.sql:18]: @jmendeza For a site with `470413` record in `audit` table, it took around 2.1s for a write API round-trip to the client (which includes the insert to audit log)
May need the QA team to double-test the impact on a big site with a larger audit table cc @sumerjabri

Comment (phuongnq) [reply src/main/resources/crafter/studio/database/upgrade/4.2.x/4.2.0.15-to-4.2.0.16.sql:18]: A direct INSERT into `audit` with the same table size of `470413` is instant (~0.007s)
Screenshot from 2024-11-08 14-11-43

**Review by jmendeza — APPROVED** [review]

**Review by sumerjabri — APPROVED** [review]

**Pull request merged**